### PR TITLE
OIDC support for logins

### PR DESCRIPTION
**Pull request by talavis**
Add support for login via OpenID Connect.

Adds a new table `Identifiers` with a mapping of `user_id` to `identifier`, intended to replace `User.username`.

On login, the system will compare the email from oidc to all identifiers in the table and log in as the corresponding user.

Adds the following configuration options:
```
OIDC_CLIENT_ID = ""
OIDC_CLIENT_SECRET = ""
OIDC_ACCESS_TOKEN_URL = ""
```

### CODE REVIEW

**Review by inaod568 — APPROVED** [review]

File: dds_web/__init__.py [hunk 1]
```diff
@@ -59,6 +61,13 @@ def create_app():
     db.init_app(app)  # Initialize database
     # ma.init_app(app)
 
+    # initialize OIDC
+    oauth.register("default_login",
```
Comment (ewels) [inline dds_web/__init__.py:65]: Can you run `black -l 100` quickly please? CI isn't added to this repo yet, but will be soon.. (same setup as `dds_cli`)

File: dds_web/user.py [hunk 2]
```diff
@@ -59,6 +61,68 @@ def login():
         return redirect(to_go_url)
 
 
+def do_login(session, identifier:str, password:str = "") -> bool:
+    """
+    Check if a user with matching identifier exists. If so, log in as that user.
+
+    TODO:
+      * Add support for passwords
```
Comment (ewels) [inline dds_web/user.py:69]: How do you mean?

**Review by ewels — APPROVED** [review]
LGTM! I haven't spent much time working with the DB back-end and stuff though, so my review isn't worth very much.

File: dds_web/user.py (hunk 2 above)
Comment (talavis) [reply dds_web/user.py:69]: I intend that function to be the way to log in, but I didn't spend the time to make it compatible with the existing password login

**Pull request merged**